### ISSUE TITLE
how to set different learning rates for different layers when fine-tuning an existing model

### ISSUE THREAD

**Issue by LinZhineng** [gpleiss/nnlr#7]
hi,
I want to fine-tune from an existing model, whether it is simple to set the learning rate of each layer? How to do?  The function "module:getOptimConfig" can not input a vector.

thanks and best regards,

**Comment by gpleiss** [comment]
You could do something like this:

```lua
local model = torch.load('my-model-path.t7')
for i = 1, model:size() - 1 do
  model:get(i):learningRate('weight', 0)
  model:get(i):learningRate('bias', 0)
  model:get(i):weightDecay('weight', 0)
  model:get(i):weightDecay('bias', 0)
end
```

That would make all the layers but the last layer fixed.
Hope this helps 😃

**Comment by LinZhineng** [comment]
thanks for your reply!
I have one question about function `model:get(index)` , which could get the index layer of the model, but which can't set the learningRate to the sub-modules in the table layer like residual block.
I define the ResUnit like following:
```
local function ResUnit(n_input_plane, n_middle_plane, n_output_plane)
   local convs = nn.Sequential()
   convs:add(nn.SpatialConvolution(n_input_plane, n_middle_plane,3,3,1,1,1,1))
   convs:add(nn.SpatialBatchNormalization(n_middle_plane))
   convs:add(nn.ReLU())
   convs:add(nn.SpatialConvolution(n_middle_plane, n_output_plane,3,3,1,1,1,1))
   convs:add(nn.SpatialBatchNormalization(n_output_plane))
   local shortcut = nn.Identity()
   return nn.Sequential()
            :add(nn.ConcatTable()
               :add(convs)
               :add(shortcut))
            :add(nn.CAddTable())
end
```
but the whole model views ResUnit as "one layer" as following, there could not locate the sub-modules, such as SpatialBatchNormalization, by model:get(index)
```
(7): nn.Sequential {
    [input -> (1) -> (2) -> output]
    (1): nn.ConcatTable {
      input
        |`-> (1): nn.Sequential {
        |      [input -> (1) -> (2) -> (3) -> (4) -> (5) -> output]
        |      (1): cudnn.SpatialConvolution(64 -> 64, 3x3, 1,1, 1,1)
        |      (2): cudnn.SpatialBatchNormalization
        |      (3): cudnn.ReLU
        |      (4): cudnn.SpatialConvolution(64 -> 64, 3x3, 1,1, 1,1)
        |      (5): cudnn.SpatialBatchNormalization
        |    }
         `-> (2): nn.Identity
         ... -> output
    }
    (2): nn.CAddTable
  }
```

**Comment by gpleiss** [comment]
If you are trying to change the LR on the first SpatialBatchNormalization layer, you could do

```lua
local resUnit = model:get(7) -- Getting the outer sequential block
local bnLayer = resUnit:get(1):get(1):get(2)
```

**Comment by LinZhineng** [comment]
thanks a lot!
